Dockerfile

```diff
@@ -38,20 +38,36 @@ RUN pip install "tensorly>=0.8.1" https://github.com/tensorly/torch/archive/715a
 # copy modulus source
 COPY . /modulus/
 
+# Install Numcodecs (This needs a separate install because Numcodecs ARM pip install has issues) 
+# A fix is being added here: https://github.com/zarr-developers/numcodecs/pull/315 but the public release is not ready yet.
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then \
+        echo "Pip install for numcodecs for $TARGETPLATFORM exists, installing!" && \
+        pip install numcodecs; \
+    elif [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/numcodecs-0.11.0-cp310-cp310-linux_aarch64.whl" ]; then \
+        echo "Numcodecs wheel for $TARGETPLATFORM exists, installing!" && \
+        pip install --force-reinstall /modulus/deps/numcodecs-0.11.0-cp310-cp310-linux_aarch64.whl; \
+    else \
+        echo "Numcodecs wheel for $TARGETPLATFORM is not present, attempting to build from pip, but might fail" && \
+	pip install numcodecs; \
+    fi
+
 # install vtk and pyvista
-RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
-	echo "Installing vtk and pyvista for: $TARGETPLATFORM" && \
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/vtk-9.2.6.dev0-cp310-cp310-linux_aarch64.whl" ]; then \
```

Comment:
Where does one get this wheel from and can this path be a dockerfile arg that can be overridden?

Reply:
Referring to my comment here: https://github.com/NVIDIA/modulus/pull/174#issuecomment-1739820835, just want to highlight that these pre-built wheels are not a necessity. In case they are unavailable, the build will attempt to install from source. I have tested this on an arm and x86 system. The motivation to add these pre-built wheels is primarily to speed-up internal builds.